### PR TITLE
FIX: Fix `merge_asof` test disabled sql simplifier

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-NNNNNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Comment by sfc-gh-nkrishna** [comment]
Kicked off daily run here: https://github.com/snowflakedb/snowpark-python/actions/runs/10529512019

**Review by sfc-gh-yzou — APPROVED** [review]

File: tests/integ/modin/test_merge_asof.py [hunk 1]
```diff
@@ -259,16 +261,18 @@ def test_merge_asof_negative_non_numeric_on(left_right_native_df_non_numeric_on)
...
+        ),
     ):
         native_pd.merge_asof(left_native_df, right_native_df, on="a")
     # Snowpark pandas raises a SnowparkSQLException
     # MATCH_CONDITION clause is invalid: The left and right side expressions must be numeric or timestamp expressions.
     with pytest.raises(
         SnowparkSQLException,
     ):
-        pd.merge_asof(left_snow_df, right_snow_df, on="a")
+        pd.merge_asof(left_snow_df, right_snow_df, on="a").to_pandas()
```
Comment (sfc-gh-yzou) [inline tests/integ/modin/test_merge_asof.py:275]: @sfc-gh-nkrishna are those test skipped in the merge gate? how com it didn't fail in the merge gate?

Comment (sfc-gh-nkrishna) [reply tests/integ/modin/test_merge_asof.py:275]: Yeah I'm not sure about that as well, it shouldn't be skipped....

Comment (sfc-gh-azhan) [reply tests/integ/modin/test_merge_asof.py:275]: They failed when the SQL optimizer was disabled which only run in github daily.

Comment (sfc-gh-yzou) [reply tests/integ/modin/test_merge_asof.py:275]: @sfc-gh-azhan do you mean SqlSimpflication?

Comment (sfc-gh-yzou) [reply tests/integ/modin/test_merge_asof.py:275]: that kind of make sense, the sql simplification tiggers an describing call when extracting ColumnState for select, that could explain it

Comment (sfc-gh-azhan) [reply tests/integ/modin/test_merge_asof.py:275]: Oh yes

**Review by sfc-gh-azhan — APPROVED** [review]

**Pull request merged**